### PR TITLE
Feat (data): compute perplexity at float32

### PR DESCRIPTION
**Pull request by i-colbert**
## Reason for this PR

Decouples model precision from evaluation precision. Allows running perplexity evaluation at a different precision (e.g., `float32`) than the model execution precision (e.g., `bfloat16`) for more accurate evaluation metrics.

## Changes Made in this PR

Added `dtype` parameter to `compute_perplexity()` in `eval.py`

## Testing Summary

No existing tests directly use `compute_perplexity`. The LLM tests currently use float32, so no updates needed.

## Risk Highlight

- [ ] This PR includes code from another work (please detail).
- [ ] This PR contains API-breaking changes.
- [ ] This PR depends on work in another PR (please provide links/details).
- [ ] This PR introduces new dependencies (please detail).
- [ ] There are coverage gaps not covered by tests.
- [ ] Documentation updates required in subsequent PR.

## Checklist

- [ ] Code comments added to any hard-to-understand areas, if applicable.
- [ ] Changes generate no new warnings.
- [ ] Updated any relevant tests, if applicable.
- [x] No conflicts with destination `dev` branch.
- [x] I reviewed my own code changes.
- [ ] Initial CI/CD passing.
- [ ] 1+ reviews given, and any review issues addressed and approved.
- [ ] Post-review full CI/CD passing.

### CODE REVIEW

File: src/brevitas_examples/llm/llm_quant/eval.py [hunk 1]
```diff
@@ -96,11 +97,12 @@ def compute_perplexity(
             # Fuse batch and sequence length dimensions.
             reference_labels = reference_labels.view(reference_labels.shape[-1])
             shift_logits = shift_logits.view(-1, shift_logits.shape[-1])
+            shift_logits = shift_logits.to(dtype)
```
Comment (Giuseppe5) [inline src/brevitas_examples/llm/llm_quant/eval.py:100]: Shouldn't we cast also reference_labels? if anything, for an abondance of precaution?

Comment (i-colbert) [reply src/brevitas_examples/llm/llm_quant/eval.py:100]: Since the loss function is cross entropy, the reference labels are integers. Casting to float32 would throw an error.
